apps/src/netsim/NetSimRouterLogModal.js

```diff
@@ -148,28 +171,193 @@ NetSimRouterLogModal.prototype.render = function () {
+    if (this.sortDescending_) {
+      insertHere = nextNew.sortValue >= nextOld.sortValue;
+    } else {
+      insertHere = nextNew.sortValue <= nextOld.sortValue;
+    }
+
+    if (insertHere) {
+      nextNew.tableRow.insertBefore(nextOld.tableRow);
+      nextNew = getNextInfo(newRows, nextNew.index + 1, getSortValue);
+    } else {
+      nextOld = getNextInfo(oldRows, nextOld.index + 1, getSortValue);
+    }
+  }
+
+  // Put whatever's left on the end of the table
+  tbody.append(newRows.slice(nextNew.index));
+};
+
+/**
+ * Generates a helper object for performing the log row merge.
+ * @param {jQuery} rows - Wrapped collection of table rows.
+ * @param {!number} atIndex - Index into `rows` at which info should be generated.
+ * @param {!function(NetSimLogEntry)} getSortValue - function to get current
+ *        sort value from log entry object.
+ * @returns {{index: number, tableRow: jQuery, sortValue: ?}}
+ */
+function getNextInfo(rows, atIndex, getSortValue) {
+  var tempRow = rows.eq(atIndex);
+  return {
+    index: atIndex,
+    tableRow: tempRow,
+    sortValue: tempRow.length > 0 ?
+        getSortValue(tempRow.data(LOG_ENTRY_DATA_KEY)) : undefined
+  };
+}
+
+/**
+ * @param {!NetSimLogEntry[]} logEntries
+ * @returns {NetSimLogEntry[]} subset of logEntries, sorted and filtered
+ *          according to the log browser's current settings.
+ */
+NetSimRouterLogModal.prototype.getSortedFilteredLogEntries = function (logEntries) {
+  // Filter entries to current log browser filter mode
+  var filteredLogEntries = this.isAllRouterLogMode_ ?
+      logEntries :
+      logEntries.filter(function (entry) {
+        return entry.nodeID === this.router_.entityID;
+      }, this);
+
+  // Sort entries according to current log browser sort setting
+  var getSortValue = NetSimRouterLogModal.sortKeyToSortValueGetterMap[this.sortBy_];
+  var sortedFilteredLogEntries = _.sortBy(filteredLogEntries, getSortValue);
+  if (this.sortDescending_) {
+    sortedFilteredLogEntries.reverse();
+  }
+  return sortedFilteredLogEntries;
 };
 
+/**
+ * Given a log entry, generate a table row that can be added to the log modal.
+ * @param {!NetSimLogEntry} logEntry
+ * @returns {Element} a tr element.
+ * @private
+ */
+NetSimRouterLogModal.prototype.makeTableRow_ = function (logEntry) {
+  var headerFields = NetSimGlobals.getLevelConfig().routerExpectsPacketHeader;
+
+  var showToAddress = headerFields.indexOf(Packet.HeaderType.TO_ADDRESS) > -1;
+
+  var showFromAddress = headerFields.indexOf(Packet.HeaderType.FROM_ADDRESS) > -1;
+
+  var showPacketInfo = headerFields.indexOf(Packet.HeaderType.PACKET_INDEX) > -1 &&
+      headerFields.indexOf(Packet.HeaderType.PACKET_COUNT) > -1;
+
+  var originNode = logEntry.getOriginNode();
+
+  var row = document.createElement('tr');
+
+  // Store the actual logEntry on the row for sorting/merging later.
+  $(row).data(LOG_ENTRY_DATA_KEY, logEntry);
+
+  var tdTimestamp = document.createElement('td');
```

Comment:
Looks like all the td fields are identical.  How about a createTd(text) helper to reduce the duplicated code?